lib/compiler-singlepass/src/codegen_x64.rs

```diff
@@ -1805,7 +1807,7 @@ impl<'a> FuncGen<'a> {
                 .collect(),
         );
 
-        let mut assembler = Assembler::new().unwrap();
+        let mut assembler = Assembler::new(0);
```

Comment:
cc @olonho, could you confirm that the change here makes sense? 

Basically: 

* `Assembler` struct writes jited code to the mmaped regions it itself maintains, which necessitates unneeded syscalls
* `VecAssembler` struct writes jitted code to a `Vec<u8>`. It needs to know the base address where the code will be loaded to properly compute some offsets
* Howerever, the relocation that wasmer actually uses is always relative to program counter: 

https://docs.rs/dynasmrt/1.1.0/src/dynasmrt/x64.rs.html#55-57

That is, we can pass zero here, and it will be essentially ignored.

Reply:
Guess it's OK

Reply:
:+1: